test/specs/events.spec.js

```diff
@@ -1,56 +1,224 @@
+        pluginOpts.annotations = [options];
 
-  });
+        const chart = window.acquireChart(chartConfig);
+        window.triggerMouseEvent(chart, 'mousemove', center(chart));
+        window.afterEvent(chart, 'mousemove', function() {
+          expect(enterSpy.calls.count()).toBe(0);
+
+          window.triggerMouseEvent(chart, 'mousemove', x0y0);
+
+          window.afterEvent(chart, 'mousemove', function() {
+            expect(leaveSpy.calls.count()).toBe(0);
+            done();
+          });
+        });
+      });
+
+      it('should not detect any move events (because only click is set)', function(done) {
+        const enterSpy = jasmine.createSpy('enter');
+        const leaveSpy = jasmine.createSpy('leave');
+        const clickSpy = jasmine.createSpy('click');
+
+        targetOptions.click = clickSpy;
+        pluginOpts.annotations = [options];
+
+        const chart = window.acquireChart(chartConfig);
+        window.triggerMouseEvent(chart, 'mousemove', center(chart));
+        window.afterEvent(chart, 'mousemove', function() {
+          expect(enterSpy.calls.count()).toBe(0);
+
+          window.triggerMouseEvent(chart, 'mousemove', x0y0);
+
+          window.afterEvent(chart, 'mousemove', function() {
+            expect(leaveSpy.calls.count()).toBe(0);
+            delete targetOptions.click;
+            done();
+          });
+        });
+      });
+
+      it('should not detect any events (because unmanaged event)', function(done) {
+        const clickSpy = jasmine.createSpy('click');
+
+        targetOptions.click = clickSpy;
+        pluginOpts.annotations = [options];
 
+        const chart = window.acquireChart(chartConfig);
+        window.afterEvent(chart, 'touchstart', function() {
+          expect(clickSpy.calls.count()).toBe(0);
+          delete targetOptions.click;
+          done();
+        });
+        window.triggerMouseEvent(chart, 'touchstart', center(chart));
+      });
+
+      it('should not call removed hook', function(done) {
+        const enterSpy = jasmine.createSpy('enter');
+        const leaveSpy = jasmine.createSpy('leave');
+
+        pluginOpts.enter = enterSpy;
+        pluginOpts.leave = leaveSpy;
+        pluginOpts.annotations = [options];
+
+        const chart = window.acquireChart(chartConfig);
+        pluginOpts.enter = undefined;
+        chart.update();
+
+        window.triggerMouseEvent(chart, 'mousemove', center(chart));
+        window.afterEvent(chart, 'mousemove', function() {
+          expect(enterSpy.calls.count()).toBe(0);
+
+          window.triggerMouseEvent(chart, 'mousemove', x0y0);
+
+          window.afterEvent(chart, 'mousemove', function() {
+            expect(leaveSpy.calls.count()).toBe(1);
+            delete pluginOpts.enter;
+            delete pluginOpts.leave;
+            done();
+          });
+        });
+      });
+
+      it('should detect a property in the context, to check persistency', function(done) {
```

Comment:
This (old) description seems misleading, its not detecting any properties. In my opinion, a better description would be "it should persist properties set in context"

Reply:
OK